isomorphe/batch.py

```diff
@@ -1,172 +1,283 @@
+        icon="🟠",
+    ),
+    RecordStatus(
+        code=SuccessTransformBatchRecord.status_code_for(needs_check=False),
+        priority=12,
+        label="Modifié",
+        icon="🟢",
+    ),
+    RecordStatus(
+        code=SkippedTransformBatchRecord.status_code_for(needs_check=True),
+        priority=3,
+        label="Ignoré, à vérifier",
+        icon="🟡",
+    ),
+    RecordStatus(
+        code=SkippedTransformBatchRecord.status_code_for(needs_check=False),
+        priority=13,
+        label="Ignoré",
+        icon="⚪️",
+    ),
+)
+
+
+class TransformBatch[R: TransformBatchRecord](Batch[R]):
+    RECORD_STATUSES = TRANSFORM_RECORD_STATUSES
+
+    def __init__(self, transformation: str, records: Sequence[R] | None = None):
         self.transformation = transformation
+        super().__init__(records)
 
-    def add(self, batch: TransformBatchRecord):
-        self.records.append(batch)
+    def filter_status(self, statuses: Sequence[int] | None = None) -> "TransformBatch[R]":
+        if statuses is None:  # not the same as []
+            return self
+        records = [r for r in self.records if r.status_code in statuses]
+        return TransformBatch[R](self.transformation, records)
 
-    def successes(self) -> list[SuccessTransformBatchRecord]:
-        return [r for r in self.records if isinstance(r, SuccessTransformBatchRecord)]
+    def filter_type[T: TransformBatchRecord](self, t: type[T]) -> "TransformBatch[T]":
+        records = [r for r in self.records if isinstance(r, t)]
+        return TransformBatch[T](self.transformation, records)
 
-    def failures(self) -> list[FailureTransformBatchRecord]:
-        return [r for r in self.records if isinstance(r, FailureTransformBatchRecord)]
+    def successes(self) -> "TransformBatch[SuccessTransformBatchRecord]":
+        return self.filter_type(SuccessTransformBatchRecord)
 
-    def skipped(self) -> list[SkippedTransformBatchRecord]:
-        return [r for r in self.records if isinstance(r, SkippedTransformBatchRecord)]
+    def failures(self) -> "TransformBatch[FailureTransformBatchRecord]":
+        return self.filter_type(FailureTransformBatchRecord)
 
-    def __repr__(self):
-        return f"TransformBatch({len(self.records)} records, {len(self.failures())} failures, {len(self.successes())} successes, {len(self.skipped())} skipped)"
+    def skipped(self) -> "TransformBatch[SkippedTransformBatchRecord]":
+        return self.filter_type(SkippedTransformBatchRecord)
 
-    # FIXME: needed?
-    def to_mef(self):
-        mef = MefArchive()
-        for r in self.records:
-            if isinstance(r, SuccessTransformBatchRecord):
-                mef.add(r.uuid, r.result, r.info)
-        return mef.finalize()
+    # TODO: drop
+    # def to_mef(self):
+    #     mef = MefArchive()
+    #     for r in self.records:
+    #         if isinstance(r, SuccessTransformBatchRecord):
+    #             mef.add(r.uuid, r.result, r.info)
+    #     return mef.finalize()
```

Comment:
Shall we drop then?

Reply:
I planned another pass to drop the MEF stuff, but yeah, let's drop that part now since it's commented-out.